lib/active_zuora/lazy_attr.rb

```diff
@@ -0,0 +1,39 @@
+module ActiveZuora
+  module LazyAttr
+
+    # This is meant to be included onto an Invoice class.
+    # Returns true/false on success.
+    # Result hash is stored in #result.
+    # If success, the id will be set in the object.
+    # If failure, errors will be present on object.
+
+    extend ActiveSupport::Concern
+
+    included do
+      include Base
+    end
+
+
+    def fetch_field(field_name)
+      return nil unless self.id
+      query_string = "select #{self.class.get_field!(field_name).zuora_name} from #{zuora_object_name} where Id = '#{self.id}'"
+      response = self.class.connection.request(:query){ |soap| soap.body = { :query_string => query_string } }
+      response[:query_response][:result][:records][field_name.to_sym]
+    end
+    private :fetch_field
+
+    module ClassMethods
+      def lazy_load(*field_names)
+        (@lazy_loadded_fields ||= []).concat field_names.map(&:to_sym)
```

Comment:
Should be `lazy_loaded_fields`. ALso, I think we maybe could simplify this by doing the following:

`@lazy_loaded_fields = Array(field_names).map(&:to_sym)`